helper/resource/id.go

```diff
@@ -25,15 +24,27 @@ func UniqueId() string {
 // Helper for a resource to generate a unique identifier w/ given prefix
 //
 // After the prefix, the ID consists of an incrementing 26 digit value (to match
-// previous timestamp output).
+// previous timestamp output).  After the prefix, the ID consists of a timestamp
+// and an 8 hex digit value which increments from 1 every time the timestamp
+// changes. The timestamp means that multiple IDs created with the same prefix
+// will sort in the order of their creation, even across multiple terraform
+// executions, as long as the clock is not turned back between calls.
 func PrefixedUniqueId(prefix string) string {
+	// Be precise to 4 digits of fractional seconds, but remove the dot before the
+	// fractional seconds.
+	timestamp := strings.Replace(
+		time.Now().UTC().Format("20060102150405.0000"), ".", "", 1)
+
 	idMutex.Lock()
 	defer idMutex.Unlock()
-	return fmt.Sprintf("%s%026x", prefix, idCounter.Add(idCounter, big.NewInt(1)))
-}
 
-func randomBytes(n int) []byte {
-	b := make([]byte, n)
-	rand.Read(b)
-	return b
+	// Reset the ID counter each time the timestamp changes. This allows 4 billion
+	// IDs every 10th of a millisecond, which ought to be good enough.
+	if timestamp != lastTimestamp {
+		lastTimestamp = timestamp
+		idCounter = 0
```

Comment:
I think we can just skip resetting this altogether, and just let it roll over if it ever gets that far. You could seed it with a pseudo random number of you want to keep it visually interesting, but that's not required ;) 

While it doesn't hurt in most cases, until go1.9 there's actually a good chance that a group of resources made in quick succession could have the timestamp bounce back again. Not to mention Windows only has a 15ms resolution.